validator/src/journal/block_manager.rs

```diff
@@ -66,6 +66,24 @@ impl BlockRef {
     pub fn block_id(&self) -> &str {
         &self.block_id
     }
+
+    pub fn get_block(&self) -> Block {
+        self.block_manager
+            .get(&[&self.block_id])
+            .next()
+            .unwrap_or_else(|| {
+                panic!(
```

Comment:
If this code bug happens, we should definitely log it, but is it truly unrecoverable? Only the calling code could determine that, not this relatively low-level code.